nipype/algorithms/confounds.py

```diff
@@ -1158,15 +1231,22 @@ def compute_noise_components(imgseries, mask_images, num_components,
     period_cut: minimum period (in sec) for DCT high-pass filter
     repetition_time: time (in sec) between volume acquisitions
```

Comment:
Maybe note that repetition_time must be defined if `filter_type` is `'cosine'`?

Similarly, it would be good to have a check for that and raise an informative exception.